crates/cli/src/commands/syn2mas.rs

```diff
@@ -0,0 +1,83 @@
+use std::process::ExitCode;
+
+use anyhow::Context;
+use clap::Parser;
+use figment::Figment;
+use mas_config::{ConfigurationSectionExt, DatabaseConfig};
+use rand::thread_rng;
+use sqlx::{Connection, Either, PgConnection};
+use syn2mas::{LockedMasDatabase, MasWriter, SynapseReader};
+use tracing::{error, warn};
+
+use crate::util::database_connection_from_config;
+
+#[derive(Parser, Debug)]
+pub(super) struct Options {
+    #[command(subcommand)]
+    subcommand: Subcommand,
+
+    /// This version of the syn2mas tool is EXPERIMENTAL and INCOMPLETE. It is only suitable for TESTING.
+    /// If you want to use this tool anyway, please pass this argument.
+    ///
+    /// If you want to migrate from Synapse to MAS today, please use the Node.js-based tool in the MAS repository.
+    #[clap(long = "i-swear-i-am-just-testing-in-a-staging-environment")]
+    experimental_accepted: bool,
+}
+
+#[derive(Parser, Debug)]
+enum Subcommand {
+    Check,
+    Migrate,
+}
+
+/// The number of parallel writing transactions active against the MAS database.
+const NUM_WRITER_CONNECTIONS: usize = 8;
+
+impl Options {
+    pub async fn run(self, figment: &Figment) -> anyhow::Result<ExitCode> {
```

Comment:
Generally, I would appreciate tracing spans on most operations, which would help a lot diagnose what takes time, but that can be introduced later

Reply:
we can always add more infos or better names on them later, but this is fine 👍